src/keys.js

```diff
@@ -33,10 +36,143 @@ export const EXTENDED_PUBLIC_KEY_VERSIONS = {
  * @throws Error with message indicating the invalid prefix.
  */
 export function validatePrefix(prefix) {
-  if (!EXTENDED_PUBLIC_KEY_VERSIONS[prefix]) throw new Error(`Invalid prefix "${prefix}" for extended public key.`);
+  if (!EXTENDED_PUBLIC_KEY_VERSIONS[prefix]) {
+    throw new Error(`Invalid prefix "${prefix}" for extended public key.`);
+  }
   return null;
 }
 
+/**
+ * Struct object for encoding and decoding extended public keys.
+ * base58 encoded serialization of the following information:
+ * [ version ][ depth ][ parent fingerprint ][ key index ][ chain code ][ pubkey ]
+ * @param {string} options.bip32Path e.g. m/45'/0'/0
+ * @param {string} options.pubkey pubkey to derive from
+ * @param {string} options.chaincode chaincode corresponding to pubkey and path
+ * @param {string} options.parentFingerprint - fingerprint of parent public key
+ * @param {string} [options.network = mainnet] - mainnet or testnet
+ * @example
+ * import { ExtendedPublicKey } from "unchained-bitcoin"
+ * const xpub = ExtendedPublicKey.fromBase58("xpub6CCHViYn5VzKSmKD9cK9LBDPz9wBLV7owXJcNDioETNvhqhVtj3ABnVUERN9aV1RGTX9YpyPHnC4Ekzjnr7TZthsJRBiXA4QCeXNHEwxLab")
+ * console.log(xpub.encode()) // returns raw Buffer of xpub encoded as per BIP32
+ * console.log(xpub.toBase58()) // returns base58 check encoded xpub
+ */
+export class ExtendedPublicKey extends Struct {
```

Comment:
Using the `Struct` interface gives us a lot of built in tools including encoding and decoding methods and a `clone` method. We're building on top of this to have a `toBase58` and `fromBase58` interface to easily retrieve the xpub given a set of data about a BIP32 node. This could be expanded upon in the future to have a more robust BIP32 interface and break dependence on the bitcoinjs implementation.